sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -271,7 +271,9 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
    */
   def apply(name: String): StructField = {
     nameToField.getOrElse(name,
-      throw new IllegalArgumentException(s"""Field "$name" does not exist."""))
+      throw new IllegalArgumentException(
+        s"""Field "$name" does not exist.
+           |Available fields: ${fieldNamesSet.mkString(", ")}""".stripMargin))
```

Comment:
shall we use `fieldNames` to keep the field order?

Reply:
done